index.js

```diff
@@ -708,27 +724,48 @@ Parser.prototype = {
     return {type: 'MixinBlock', line: tok.line, filename: this.filename};
   },
 
+  parseYield: function() {
+    var tok = this.expect('yield');
+    return {type: 'YieldBlock', line: tok.line, filename: this.filename};
+  },
+
   /**
    * include block?
    */
 
   parseInclude: function(){
     var tok = this.expect('include');
-
-    return {
+    var node = {
       type: 'Include',
       file: {
         type: 'FileReference',
-        path: tok.val.trim(),
         line: tok.line,
         filename: this.filename
       },
-      filter: tok.filter,
-      attrs: tok.attrs ? tok.attrs.attrs : [],
-      block: 'indent' == this.peek().type ? this.block() : this.emptyBlock(tok.line),
       line: tok.line,
       filename: this.filename
     };
+    var filters = [];
+    while (this.peek().type === 'filter') {
+      filters.push(this.parseIncludeFilter());
+    }
+    var path = this.expect('path');
+
+    node.file.path = path.val.trim();
+
+    if (/\.jade$/.test(node.file.path) && !filters.length) {
+      node.block = 'indent' == this.peek().type ? this.block() : this.emptyBlock(tok.line);
+    } else {
+      node.type = 'RawInclude';
+      node.filters = filters;
+      if (this.peek().type === 'indent') {
+        // If there is a block, just ignore it.
+        this.block();
+        // TODO: make this a warning
+        // this.error('Raw inclusion cannot contain a block; block ignored', 'RAW_INCLUDE_BLOCK', this.peek());
```

Comment:
Why not just make this a hard error?  The block has always been ignored in the past so it seems fine to remove it as we release this major version.